File: C-programs-and-solutions/Programs and executables/02_lsf/lsf.c
//
//  lsf.c
//
//  Created by V.H. Belvadi on 02/03/17.
//

#include<stdio.h>
#include<stdlib.h>
#include<math.h>
int main()
{
    FILE *input;
    int n,i,x[20],y[20],sumx=0,sumy=0,sumxy=0,sumx2=0;
    float m,c;
    char name[20];
    
    system("clear");
    
    printf("\n\nEnter the name of the input file: \t");
    scanf("%s", name);
    
    printf("\n\nEnter the number of rows of data:\t");
    scanf("%d",&n);
    
    input = fopen(name,"r");
    
    for(i=0; i<n; i++)
        fscanf(input,"%d\t%d\n",&x[i],&y[i]);
    
    fclose(input);
    
    for(i=0;i<n;i++)
    {
        sumx=sumx +x[i];
        sumx2=sumx2 +x[i]*x[i];
        sumy=sumy +y[i];
        sumxy=sumxy +x[i]*y[i];
        
    }
    
    c=((sumx2*sumy -sumx*sumxy)*1.0/(n*sumx2-sumx*sumx)*1.0);
    m=((n*sumxy-sumx*sumy)*1.0/(n*sumx2-sumx*sumx)*1.0);
    printf("\n\nThe line of best fit is y=%3.3fx +%3.3f \n\n",m,c);
    
    return 0;
}

File: PERL-programs-and-solutions/Programs and executables/06_lsf/lsf.pl
#! /usr/bin/perl -w

print "\nEnter the name of the input file:\t";
$file = <>;
print "Enter the number of rows of data:\t";
$rows = <>;

open (INPUT,$file) or die "Could not open the file";
# @line;
@arrayX = ();
@arrayY = ();
$counter = 0;

($i, $sumx, $sumy, $sumxy, $sumx2) = (0, 0, 0, 0, 0);

while(<INPUT>){
    @line = split(/\t/,$_);
    $arrayX[$counter]=$line[0];
    $arrayY[$counter]=$line[1];
    $counter++;
}

for($i=0;$i<$rows;$i++)
    {
        $sumx = $sumx + $arrayX[$i];
        $sumx2 = $sumx2 + $arrayX[$i]*$arrayX[$i];
        $sumy = $sumy + $arrayY[$i];
        $sumxy = $sumxy + $arrayX[$i]*$arrayY[$i];
        
    }
    
$c = (($sumx2*$sumy - $sumx*$sumxy)*1.0/($rows*$sumx2-$sumx*$sumx)*1.0);
$m = (($rows*$sumxy-$sumx*$sumy)*1.0/($rows*$sumx2-$sumx*$sumx)*1.0);
print "\n\nThe line of best fit is y=$m x + $c \n\n";
